### PR TITLE
Calculate movement method, wrangler and tests

### PR DESCRIPTION
**Pull request by Jordancooke**

### CODE REVIEW

**Comment by NigelBond** [comment]
We think this code is incorrect 
`
                # This check is too prevent the DivdebyZeroError.
                if current_list[i] and previous_list[i] != 0:
                    number = (current_list[i] - previous_list[i]) / current_list[i]
                else:
                    number = 0.0
`
We think it should be (current - previous) / previous not current

**Comment by NigelBond** [comment]
We also think the same segment of code is incorrect where is sets number = 0.0

                if current_list[i] and previous_list[i] != 0:
                    number = (current_list[i] - previous_list[i]) / current_list[i]
                else:
                    number = 0.0

**Comment by NigelBond** [comment]
Reopening

**Comment by NigelBond** [comment]
Not full test coverage for the calculate_movement wrangler and method, particulary exception cases.

**Comment by NigelBond** [comment]
There's a few lines of code in the wrangler that are commented out, which perhaps could be removed (line 164 at the moment)

# sqs.purge_queue(
        #     QueueUrl=queue_url
        # )

**Comment by Jordancooke** [comment]
@NigelBond 

For some strange reason I can't seem to reply to your comments or mark them as resolved, so I will collate my replies and place them here...

1: Good spot, I have updated the code to reflect you correction.

2: Mike pointed this issue out to me before and I made a note to change it, but, that note seems to have gone walkies. Changes have been made to prevent this problem for occurring.

3: As test coverage goes I believe it's around 87% for the wrangler and 91% for the method. I believe this would fall under sufficient coverage. I will cover exceptions in the tests as technical debt.

4: The purge functionality is currently commented out as while we are in pre-production this feature is more of a hinderance than useful, when it come to fixing errors. Also an issue with the purge command and the limit of 60 seconds between purges is still to have a solution implemented so alternative measures will likely be implemented. The code has been removed until a decision is made.

**Comment by NigelBond** [comment]
Jolly good

**Comment by Jordancooke** [comment]
@bitmonkey The changes suggested have been implemented. The flake8 errors which remain are either required by the code e.g. the import's being below a line of code, which is setting it's location or are flake8's configuration being set to 79 instead of 90.

**Comment by bitmonkey** [comment]
@Jordancooke 

> @bitmonkey The changes suggested have been implemented. The flake8 errors which remain are either required by the code e.g. the import's being below a line of code, which is setting it's location or are flake8's configuration being set to 79 instead of 90.

I've updated the docker environment so you can get rid of the `sys.path` hack and flake8 won't complain about the imports, we now set `PYTHONPATH` when we run the tests so it will look for files in the source directory.

The tests don't run though, there's a syntax error in `test_calculate_movement` which prevents the run.

[test_movement_logs.txt](https://github.com/ONSdigital/es-imputation-sg/files/3440756/test_movement_logs.txt)

You still have some lines > 90 too:

[calc_movement_lint.txt](https://github.com/ONSdigital/es-imputation-sg/files/3440764/calc_movement_lint.txt)

**Comment by Jordancooke** [comment]
@bitmonkey 
> I've updated the docker environment so you can get rid of the `sys.path` hack and flake8 won't complain about the imports, we now set `PYTHONPATH` when we run the tests so it will look for files in the source directory.
> 
>The tests don't run though, there's a syntax error in `test_calculate_movement` which prevents the run.

I have pulled down your changes to the docker environment and removed the sys line from the imports and solved any remaining flake8 problems associated with movements. I have also fixed the broken assert in test_calculate_movement.

**Comment by kingmushroom** [comment]
Most changes in place. wrangler is stuck at 99% coverage(the marshmallow bit) because the error message uses checkpoint, which comes from an env variable. This is fixed in the error handling branch that'll follow.

File: calculate_movement_wrangler.py [hunk 1]
```diff
@@ -25,88 +53,112 @@ def _get_traceback(exception):
...
-        # Setting up environment variables
-        s3_file = os.environ['s3_file']
-        bucket_name = os.environ['bucket_name']
-
-        queue_url = os.environ['queue_url']
-        sqs_messageid_name = os.environ['sqs_messageid_name']
+        schema = EnvironSchema()
+        config, errors = schema.load(os.environ)
+        if errors:
+            raise ValueError(f"Error validating environment params: {errors}")
```
[the remark this answers is not in the document]
Comment (bitmonkey) [reply calculate_movement_wrangler.py]: Agreed with Mike to leave this untested, this is changed in the error handling branch.

**Pull request merged**